### DISCUSSION TITLE
Replace Next.JS/Gatsby

### DISCUSSION THREAD

**Discussion by dioptre** [parcel-bundler/parcel#7086] [Ideas]
With Parcel v2 and the new build/optimization processes it makes sense to introduce a static site generation middleware to replace probably all of NextJS/Gatsby. 

Has any work been done on this? Wondering if there would be a good place to start on this?

I am getting massive memory leaks on Gatsby with tens of thousands of pages, and looking to develop a solution asap. Would be great to know if other people have thought about this too.

Andrew

**Comment by DeMoorJasper** [comment]
Yes Parcel 2 does unlock a lot of potential in this field.

The only open-source project I know of that uses some kind of static site generation with parcel v2 is the docs site of react spectrum https://github.com/adobe/react-spectrum/tree/main/packages/dev

Replacing gatsby or nextjs isn't that easy though, they have immense amounts of funding and dedicated teams.

**Reply by dioptre** [reply]
Sounds like we should get some funding for Parcel :)

**Reply by dioptre** [reply]
@devongovett can you weigh in on this? Seems like you wrote the work for Adobe?

**Comment by skipjack** [comment]
What happened with this tweet? @devongovett is that plugin you mention there open source now?

https://twitter.com/devongovett/status/1226954310161821696

I'd love to be able to easily pre-render a couple static pages within an SPA. It almost looks like we could use an `.mdx` file rather than a `.html` file as our entry point?
